Libraries/HyperV.psm1

```diff
@@ -664,17 +664,25 @@ function Apply-HyperVCheckpoint {
         $VMData,
         [string]$CheckpointName
     )
-
     foreach ($VM in $VMData) {
-        Stop-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost -TurnOff -Force
-        Restore-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ComputerName $VM.HyperVHost -Confirm:$false
-        $msg = ("VM:{0} restored to checkpoint: {1}" `
+        Get-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ErrorAction Ignore | out-null
+        if($?) {
+            Stop-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost -TurnOff -Force
+            Restore-VMSnapshot -Name $CheckpointName -VMName $VM.RoleName -ComputerName $VM.HyperVHost -Confirm:$false
+            $msg = ("VM:{0} restored to checkpoint: {1}" `
                  -f ($VM.RoleName,$CheckpointName))
-        Write-LogInfo $msg
-        Start-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost
+            Write-LogInfo $msg
+            Start-VM -Name $VM.RoleName -ComputerName $VM.HyperVHost
+            return $true
```

Comment:
this will return on the first VM, which changes the behavior of multiple VM scenarios

Reply:
moved this line to end of the function which will return $true if multiple VMs provided are having same  checkpoint